src/libs/http/http-deepl-client/src/lib/deepl-client.ts

```diff
@@ -33,260 +35,124 @@ export class DeeplClient {
     this.context = context;
   }
 
-  /**
-   * Retrieves the singleton instance of DeeplService.
-   * @returns {DeeplClient} The singleton instance.
-   */
-  public static async getInstanceAsync(
-    context: vscode.ExtensionContext
-  ): Promise<DeeplClient> {
-    const currentApiKey = DeeplClient.getApiKey();
-
-    if (!DeeplClient.instance || DeeplClient.previousApiKey !== currentApiKey) {
-      DeeplClient.instance = new DeeplClient(context);
-      DeeplClient.previousApiKey = currentApiKey;
-      await DeeplClient.initializeTranslator(currentApiKey);
-    }
-
-    return DeeplClient.instance;
-  }
-
-  public async fetchTranslations(
-    text: string[],
+  public async translateAsync(
+    texts: string[],
+    sourceLanguage: SourceLanguageCode,
     requestedTargetLanguage: string
   ): Promise<string[]> {
-    try {
-      if (!DeeplClient.instance.translator) {
-        throw new Error('Translator not initialized.');
-      }
+    // Mocking the translation for now
+    // return texts.map(
+    //   text => `Translated (${text}) to ${requestedTargetLanguage}`
+    // );
 
-      let targetLanguage = requestedTargetLanguage;
```

Comment:
Restore this and all related changes for fixing source and target language. But extract them to utils.